osg_configure/configure_modules/gratia.py

```diff
@@ -618,89 +426,72 @@ def _configure_htcondor_ce_probe(self):
-                    config_location, data_folder),
-                             level=logging.ERROR)
-                    valid = False
-                elif not os.path.exists(history_dir):
-                    self.log("Condor PER_JOB_HISTORY_DIR %s points to a nonexistant location" % history_dir,
-                             level=logging.ERROR)
-                    valid = False
-                else:
-                    try:
-                        if not os.path.samefile(data_folder, history_dir):
-                            self.log("DataFolder setting in %s (%s) and condor PER_JOB_HISTORY_DIR %s "
-                                     "do not match, these settings must match!" % (config_location,
-                                                                                   data_folder,
-                                                                                   history_dir),
-                                     level=logging.ERROR)
-                            valid = False
-                    except OSError as e:
-                        self.log("Error comparing DataFolder setting in %s (%s) and condor PER_JOB_HISTORY_DIR %s:\n%s"
-                                 % (config_location, data_folder, history_dir, e),
-                                 level=logging.ERROR)
-                        valid = False
-
-            # Per Gratia-126 DataFolder must end in / otherwise gratia won't find certinfo files
-            if not data_folder.endswith('/'):
-                self.log("DataFolder setting in %s must end in a /" % config_location,
-                         level=logging.ERROR)
-                valid = False
+        if not match:
+            return True
 
-        return valid
+        data_folder = match.group(1)
+        data_folder = data_folder.strip('" \t')
+        # Per Gratia-126 DataFolder must end in / otherwise gratia won't find certinfo files
+        if not data_folder.endswith('/'):
+            self.logger.error("DataFolder setting in %s must end in a /", config_location)
+            return False
+
+        history_dir = self._get_condor_ce_history_dir()
+        if not history_dir:
+            self.logger.error(textwrap.fill(
+                """Could not verify DataFolder correctness: unable to get PER_JOB_HISTORY_DIR
+                for the schedd. This may be caused by PER_JOB_HISTORY_DIR not being defined."""
+            ))
+            return False
+
+        # os.path.samefile will die if the paths don't exist so check that explicitly (SOFTWARE-1735)
+        if not os.path.exists(data_folder):
+            self.logger.error("DataFolder setting in %s (%s) points to a nonexistent location",
+                              config_location, data_folder)
+            return False
+        elif not os.path.exists(history_dir):
+            self.logger.error("condor-ce schedd's PER_JOB_HISTORY_DIR (%s) points to a nonexistent location", history_dir)
+            return False
+        else:
```

Comment:
The if/elif seems kinda goofy since you're running `os.path.samefile` in a try block below. If the paths don't exist, is the exception from `os.path.samefile` not helpful? It doesn't hurt to keep these in, though

Reply:
OK, I have 2 versions, let me know which one you prefer:
1. https://github.com/opensciencegrid/osg-configure/pull/122/commits/e537db0daedaa506f6fadac35271c1a51c873692
2. https://github.com/opensciencegrid/osg-configure/pull/122/commits/d36293dac0f3c0103b1a66aba3cf3550a7f7291f

Reply:
Let's go with 1. since you've gone through all the trouble :)